numpyro/distributions/continuous.py

```diff
@@ -1188,6 +1191,169 @@ def infer_shapes(
+                " if you need higher dimensional W.",
+            )
+        if not (isinstance(W, np.ndarray) or _is_sparse(W)):
+            raise ValueError(
+                "W needs to be a numpy array or a scipy sparse matrix. Please make a feature",
+                " request if you need to support jax ndarrays.",
+            )
+        if self.is_sparse:
+            # TODO: look into future jax sparse csr functionality and other developments
+            self.W = _to_sparse(W)
+        else:
+            assert not _is_sparse(
+                W
+            ), "W is a sparse matrix so please specify `is_sparse=True`."
+            # TODO: look into static jax ndarray representation
+            self.W = W
+
+        batch_shape = lax.broadcast_shapes(
+            jnp.shape(loc)[:-1],
+            jnp.shape(alpha),
+            jnp.shape(tau),
+            jnp.shape(self.W)[:-2],
+        )
+        event_shape = jnp.shape(self.W)[-1:]
+        (self.loc,) = promote_shapes(loc, shape=batch_shape + event_shape)
+        self.alpha, self.tau = promote_shapes(alpha, tau, shape=batch_shape)
+
+        super(CAR, self).__init__(
+            batch_shape=batch_shape,
+            event_shape=event_shape,
+            validate_args=validate_args,
+        )
+
+        if self._validate_args:
+            assert (
+                self.W.sum(axis=-1) > 0
+            ).all() > 0, "all sites in adjacency matrix W must have neighbours"
+
+            if self.is_sparse:
+                assert (self.W != self.W.T).nnz == 0, "W must be symmetric"
+            else:
+                assert np.array_equal(self.W, self.W.T), "W must be symmetric"
+
+    def sample(self, key, sample_shape=()):
+        # TODO: look into a sparse sampling method
+        mvn = MultivariateNormal(self.mean, precision_matrix=self.precision_matrix)
+        return mvn.sample(key, sample_shape=sample_shape)
+
+    @validate_sample
+    def log_prob(self, value):
+        phi = value - self.loc
+        W = self.W
+
+        if self.is_sparse:
+            D = np.asarray(W.sum(axis=-1)).squeeze(axis=-1)
+            D_rsqrt = D ** (-0.5)
+
+            W_scaled = W.multiply(D_rsqrt).multiply(D_rsqrt[:, np.newaxis]).toarray()
+
+            W = BCOO.from_scipy_sparse(W)
+
+        else:
+            D = W.sum(axis=-1)
+            D_rsqrt = D ** (-0.5)
+
+            W_scaled = W * (D_rsqrt * D_rsqrt[:, np.newaxis])
+
+        # TODO: look into sparse eignvalue methods
+        lam = np.linalg.eigvalsh(W_scaled)
+
+        n = D.shape[-1]
+
+        logtau = n * jnp.log(self.tau)
+        logdet = jnp.log1p(-jnp.expand_dims(self.alpha, -1) * lam).sum(-1)
+
+        logquad = self.tau * jnp.sum(
+            phi
+            * (
+                D * phi
+                - jnp.expand_dims(self.alpha, -1)
+                * (W @ phi[..., jnp.newaxis]).squeeze(axis=-1)
+            ),
+            -1,
+        )
+
+        return -0.5 * (logtau + logdet + logquad)
```

Comment:
```
-        return -0.5 * (logtau + logdet + logquad)
+        return 0.5 * (-n * jnp.log(2 * jnp.pi) + logtau + logdet - logquad)
```

Reply:
Are we sure `jnp.log(2*jnp.pi)` is the only constant? In the [stan article](https://mc-stan.org/users/documentation/case-studies/mbjoseph-CARStan.html), they also drop `0.5 * log(c)` from the determinant term.

Do the constants matter?

Reply:
Yes, we need `0.5 * log(c)` (see the above comment at `logdet` term).